### ISSUE TITLE
flatten encoder level

### ISSUE THREAD

**Issue by jcbyts** [NeuroTheoryUMD/NDNT#4]
Currently NDN is a default class and the model lives in `NDN.encoder`

We should flatten that level so NDN is an `nn.Module` and the encoder level doesn't exist.

**Comment by jcbyts** [comment]
@dbutts the only issue with flattening out the encoder so that NDN is the model, is that we have overloaded the method `train`, which in `nn.Modules` is a magic word and sets the model in training mode. I propose we use `NDN.fit`

**Comment by jcbyts** [comment]
Merged into main
